backend/services/market_data_service_waterfall.py

```
@@ -373,46 +332,79 @@ def get_verification_stats(self):
-                             params={'apikey': self.providers['fmp']['key']}, timeout=5)
-            data = r.json()
-            if data:
-                self.providers['fmp']['limiter'].increment()
-                return float(data[0]['price'])
+            url = f"https://financialmodelingprep.com/api/v3/quote/{symbol}?apikey={self.providers['fmp']['key']}"
+            resp = requests.get(url, timeout=5)
+            if resp.status_code == 200:
+                data = resp.json()
+                if data:
+                    return float(data[0]['price'])
             return None
```

Ruff (0.15.2)

[warning] Consider moving this statement to an `else` block

(TRY300)